### PR TITLE
BUG: Add tests for data calls

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: app/pages/capabilities-selector/controller.test.js [hunk 1]
```diff
@@ -6,8 +6,42 @@ jest.mock('../../apiProvider', () => ({
...
+
+    it('should call getData once with the correct params', async () => {
+      apiProvider.getData
+        .mockResolvedValueOnce(mockedCapabilitiesData);
+
+      await getCapabilitiesContext();
+      expect(apiProvider.getData.mock.calls.length).toEqual(1);
+      expect(apiProvider.getData.mock.calls[0].length).toEqual(1);
+      expect(Object.keys(apiProvider.getData.mock.calls[0][0]).length).toEqual(1);
+      expect(apiProvider.getData.mock.calls[0][0].endpointLocator).toBe('getCapabilities');
```
Comment (alibombaye) [inline app/pages/capabilities-selector/controller.test.js]: You are going to have to explain this one to me on Monday

Comment (kat1906) [reply app/pages/capabilities-selector/controller.test.js]: so `apiProvider.getData.mock.calls` is an array of the calls made to the mock
`apiProvider.getData.mock.calls[0]` is an array of the args used when the mock was called
`apiProvider.getData.mock.calls[0][0]` is one of the args

Comment (kat1906) [reply app/pages/capabilities-selector/controller.test.js]: https://jestjs.io/docs/en/mock-functions.html

Comment (kat1906) [reply app/pages/capabilities-selector/controller.test.js]: so it just checks that the getData function is being called with the correct args (which is what changed and made the bug but all the tests still passed - if it happens again the tests will fail)

Comment (alibombaye) [reply app/pages/capabilities-selector/controller.test.js]: Right I get this now. It kinda looks pretty difficult to get you head round looking into it the first time. Can anything be done to make it read a little better. I noticed in the link you sent you can use Custom Matchers 
`// The mock function was called at least once with the specified args
expect(mockFunc).toHaveBeenCalledWith(arg1, arg2);`
can something like this be used to check getData is being called with the correct args?

Comment (kat1906) [reply app/pages/capabilities-selector/controller.test.js]: ooh yeah that would be better - let me look a bit more

File: app/pages/capabilities-selector/controller.test.js [hunk 2]
```diff
@@ -6,8 +6,40 @@ jest.mock('../../apiProvider', () => ({
...
+      apiProvider.getData.mockReset();
+    });
+
+    it('should call getData once with the correct params', async () => {
+      apiProvider.getData
+        .mockResolvedValueOnce(mockedCapabilitiesData);
+
+      await getCapabilitiesContext();
+      expect(apiProvider.getData.mock.calls.length).toEqual(1);
+      expect(apiProvider.getData).toHaveBeenCalledWith({ endpointLocator: 'getCapabilities' });
```
Comment (alibombaye) [inline app/pages/capabilities-selector/controller.test.js]: Lovely 😍

File: app/pages/solutions-list/controller.test.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+      await getSolutionsForSelectedCapabilities({ capabilitiesSelected: 'C1+C2' });
+
+      expect(createContext.createSolutionListPageContext.mock.calls.length).toEqual(1);
+      expect(createContext.createSolutionListPageContext).toHaveBeenCalledWith({
+        filterType: 'capabilities-selector',
+        capabilitiesSelected: ['C1', 'C2'],
+        solutionListManifest: capabilitiesSelectorContent,
+        solutionsData: mockedSolutionData.solutions,
+      });
```
Comment (alibombaye) [inline app/pages/solutions-list/controller.test.js]: I really like these 👍

**Pull request merged**